TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/TrafficReplayer.java

```diff
@@ -285,27 +245,41 @@ public ReplayEngine(Consumer<RequestResponsePacketPair> fullDataHandler) {
         }
 
         @Override
-        public void accept(ReplayEntry e) {
-            if (e.operation.equals(Operation.Close)) {
-                publishAndClear(e.id);
-            } else if (e.operation.equals(Operation.Write)) {
-                var priorBuffers = liveStreams.get(e.id);
-                if (priorBuffers == null) {
+        public void accept(String id, TrafficObservation observation) {
+            boolean updateFirstResponseTimestamp = false;
+            RequestResponsePacketPair runningList = null;
+            var pbts = observation.getTs();
+            var timestamp = Instant.ofEpochSecond(pbts.getSeconds(), pbts.getNanos());
+            if (observation.hasEndOfMessageIndicator()) {
+                publishAndClear(id);
+            } else if (observation.hasRead()) {
+                runningList =
+                        liveStreams.putIfAbsent(id, new RequestResponsePacketPair());
+                if (runningList==null) {
+                    runningList = liveStreams.get(id);
+                    // TODO - eliminate the byte[] and use the underlying nio buffer
+                    runningList.addRequestData(timestamp, observation.getRead().getData().toByteArray());
+                }
+            } else if (observation.hasReadSegment()) {
+                throw new RuntimeException("Not implemented yet.");
+            } else if (observation.hasWrite()) {
+                updateFirstResponseTimestamp = true;
+                runningList = liveStreams.get(id);
+                if (runningList == null) {
                     throw new RuntimeException("Apparent out of order exception - " +
                             "found a purported write to a socket before a read!");
                 }
-                priorBuffers.addResponseData(e.timestamp, e.data);
-            } else if (e.operation.equals(Operation.Read)) {
-                var runningList =
-                        liveStreams.putIfAbsent(e.id, new RequestResponsePacketPair(e.timestamp));
-                if (runningList==null) {
-                    runningList = liveStreams.get(e.id);
-                } else if (runningList.lastOperation == Operation.Write) {
-                    publishAndClear(e.id);
-                    accept(e);
-                    return;
+                runningList.addResponseData(timestamp, observation.getWrite().toByteArray());
+            } else if (observation.hasWriteSegment()) {
+                updateFirstResponseTimestamp = true;
+                throw new RuntimeException("Not implemented yet.");
+            } else if (observation.hasRequestReleasedDownstream()) {
```

Comment:
This is to signify that we have finished forwarding request to cluster, correct. Do we set this currently in the LoggingProxy?

Reply:
Yes and no.  Here's the [commit ](https://github.com/opensearch-project/opensearch-migrations/commit/1c9e0e20c9dffd7dd440a80a96b6d8225dafb388) that created the new observation